File: py/7.recursive.py
def to_one(n):
    if n < 0:
        return
    print(n, end=' ')
    return to_one(n-1)


def sum_n(n):
    if n == 1:
        return 1
    return n + sum_n(n-1)


def fibonacci(n):
    if n < 2:
        return 1
    return fibonacci(n-1) + fibonacci(n-2)


n = int(input())

print("to_one : ", end='')
to_one(n)
print()

print("sum : " + str(sum_n(n)))

print("fibonacci : " + str(fibonacci(n)))

File: cpp/7.recursive.cpp
#include <stdio.h>

void   to_one(int n);
int     sum_n(int n);
int fibonacci(int n);

int main(){
    int n;
    scanf("%d", &n);

    printf("to_one : ");
    to_one(n);
    printf("\n");

    printf("sum : %d\n", sum_n(n));

    printf("fibonacci : %d", fibonacci(n));

    return 0;
}

void to_one(int n){
    if(n < 0)
        return;

    printf("%d ", n);
    return to_one(n-1);
}

int sum_n(int n){
    if(n == 1)
        return 1;
    return n + sum_n(n-1);
}

int fibonacci(int n){
    if(n < 2)
        return 1;
    return fibonacci(n-1) + fibonacci(n-2);
}